server/handlers/subscription_handler.coffee

```diff
@@ -281,8 +281,10 @@ class SubscriptionHandler extends Handler
 
     else
       couponID = user.get('stripe')?.couponID
-      if user.get('country') is 'brazil'
-        couponID ?= 'brazil'
+      if not couponID and user.get 'country'
+        basicProduct = yield Product.findBasicSubscriptionForUser user
+        if basicProduct.name isnt 'basic_subscription'  # We have a customized product for this country
+          couponID = user.get 'country'
```

Comment:
`yield` won't work outside of a co-routine. This is so deeply nested in handler callback hell that for it to work, either the whole endpoint would need to be refactored (ideal, but time consuming) or just use the Promise that `findBasicSubscriptionForUser` should return (which it currently doesn't but that's an easy fix).